packages/braavos-connector/src/index.ts

```diff
@@ -0,0 +1,110 @@
+import { AbstractConnector } from '@web3-starknet-react/abstract-connector';
+import { ConnectorUpdate } from '@web3-starknet-react/types';
+import { AccountInterface } from 'starknet';
+import { IStarknetWindowObject } from 'get-starknet';
+
+export class NoStarknetProviderError extends Error {
+  public constructor() {
+    super();
+    this.name = this.constructor.name;
+    this.message = 'No Starknet Provider was found on window object';
+  }
+}
+
+export class BraavosConnector extends AbstractConnector {
+  public async activate(): Promise<ConnectorUpdate> {
+    const wallet = this.getBraavosWallet();
+
+    if (!wallet) {
+      throw new NoStarknetProviderError();
+    }
+
+    wallet.on('accountsChanged', this.handleAccountsChanged);
+
+    let account: AccountInterface | undefined = wallet.account;
+    let connectedAddress: string | undefined = wallet.selectedAddress;
+
+    if (!account) {
```

Comment:
```suggestion
    if (!connectedAddress) {
```

I was also working on this package and saw that checking for `account` doesn't work properly because I think in braavos, account is an object/interface which can have selectedAddress as empty string when not connected. 

So let's check for connectedAddress